### PR TITLE
Solitaire: Confirm ending the current game in more situations

### PR DESCRIPTION
**Pull request by AtkinsSJ**
We previously had a warning when trying to exit Solitaire or Spider while a game was in progress. This adds the same functionality to other actions that would end the current game: Starting a new one, or changing the game's configuration. When changing the configuration, we do apply the setting, so it will take effect for the next game that is started.

### CODE REVIEW

File: Userland/Games/Solitaire/main.cpp [hunk 1]
```diff
@@ -145,29 +145,38 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
                 "Game in progress"sv,
                 GUI::MessageBox::Type::Warning,
                 GUI::MessageBox::InputType::YesNo);
 
             if (result == GUI::MessageBox::ExecResult::Yes)
-                return GUI::Window::CloseRequestDecision::Close;
-            else
-                return GUI::Window::CloseRequestDecision::StayOpen;
+                return true;
+            return false;
```
Comment (trflynn89) [inline Userland/Games/Solitaire/main.cpp:159]: :thinking: 
```c++
return result == GUI::MessageBox::ExecResult::Yes;
```

File: Userland/Games/Spider/main.cpp [hunk 2]
```diff
@@ -195,22 +195,27 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
                 "Game in progress"sv,
                 GUI::MessageBox::Type::Warning,
                 GUI::MessageBox::InputType::YesNo);
 
             if (result == GUI::MessageBox::ExecResult::Yes)
-                return GUI::Window::CloseRequestDecision::Close;
-            else
-                return GUI::Window::CloseRequestDecision::StayOpen;
+                return true;
+            return false;
```
Comment (trflynn89) [inline Userland/Games/Spider/main.cpp:209]: same

File: Userland/Games/Solitaire/main.cpp (hunk 1 above)
Comment (AtkinsSJ) [reply Userland/Games/Solitaire/main.cpp]: Oh yeah, that's way better!

**Comment by AtkinsSJ** [comment]
Made the suggested changes.

**Pull request merged**